### PR TITLE
Fix for #2096 "core: limit message queue size"

### PR DESCRIPTION
**Pull request by Darredevil**
Changed from unsigned int to uint16_t so it will not be dependent on the platform implementation.

### CODE REVIEW

**Comment by OlegHahm** [comment]
Please prefix your commits with the affected module name (`core` in this case) and a colon.

File: core/include/cib.h [hunk 1]
```diff
@@ -44,9 +46,9 @@ typedef struct {
  *
  * @param[out] cib      Buffer to initialize.
  *                      Must not be NULL.
- * @param[in]  size     Size of the buffer, must not exceed MAXINT/2.
+ * @param[in]  size     Size of the buffer, must not exceed 65,534.
```
Comment (Kijewski) [inline core/include/cib.h]: s/65,534/32,768/, the highest power of 2 that fits into a uint16_t.

Comment (haukepetersen) [inline core/include/cib.h]: no, 65535 would be the correct value here.2^16 = 65536 and so uint16_t is from 0 to 65535, so 65535 would be the maximum value, right?

Comment (Kijewski) [inline core/include/cib.h]: This size value fills a bit mask. 0 is treated specially as an empty CIB. So the maximum value is 2**15.

**Pull request closed without merging**